File: test/cpp/Source/UpdateLeafFlag.cpp
#include <iostream>
#include <vector>
#include<string>
#include<algorithm>
#include<cmath>
#include<time.h>
#include "UpdateLeafFlag.h"
using namespace std;

UpdateLeafFlag calculate_UpdateLeafFlag(double cumulTT,double leafNumber, vector<string> calendarMoments,vector<time_t> calendarDates,vector<double> calendarCumuls,time_t currentdate,double FinalLeafNumber,int HasFlagLeafLiguleAppeared,double phase)
{

if (phase >= 1 && phase< 4)
{
    if (leafNumber > 0)
    {
        if (HasFlagLeafLiguleAppeared == 0 && (FinalLeafNumber > 0 && leafNumber >= FinalLeafNumber))
        {
            HasFlagLeafLiguleAppeared = 1;
            size_t pos = find(calendarMoments.begin(), calendarMoments.end(),"FlagLeafLiguleJustVisible" ) - calendarMoments.begin();


            if  (pos>=calendarMoments.size())
            {
                calendarMoments.push_back("FlagLeafLiguleJustVisible");
                calendarCumuls.push_back(cumulTT);
                calendarDates.push_back(currentdate);
            }
        }
    }
    else
        HasFlagLeafLiguleAppeared = 0;
}
return UpdateLeafFlag {HasFlagLeafLiguleAppeared, calendarMoments,calendarDates,calendarCumuls};
}

File: test/f90/updateLeafFlag.f90
module updateLeafFlagModule
    use crop2mlModules
    implicit none

contains
    subroutine calculateUpdateLeafFlag( cumulTT, leafNumber,&
        calendarMoments, calendarDates, calendarCumuls,currentdate,&
        FinalLeafNumber, HasFlagLeafLiguleAppeared, phase)

        ! DECLARATION
        real, intent(in)::phase, cumulTT, leafNumber, FinalLeafNumber
        character(len=65), intent(in) :: currentdate

        real,dimension (:), intent(inout), allocatable:: calendarCumuls
        character(len=65),dimension(:), intent(inout), allocatable:: calendarMoments, calendarDates

        integer, intent(inout):: HasFlagLeafLiguleAppeared


        !Fortran Snippets
        if ((phase >= 1) .AND. (phase< 4)) then

            if (leafNumber > 0) then

                if ((HasFlagLeafLiguleAppeared == 0) .AND. (FinalLeafNumber > 0) .AND. (leafNumber >= FinalLeafNumber)) then

                    HasFlagLeafLiguleAppeared = 1;
                    if  (ALL(calendarMoments/="FlagLeafLiguleJustVisible")) then

                        call AddToListChar(calendarMoments,"FlagLeafLiguleJustVisible")
                        call AddToList(calendarCumuls,cumulTT)
                        call AddToListChar(calendarDates, currentdate)
                    end if
                end if

            else
                HasFlagLeafLiguleAppeared = 0
            end if
        end if


    end subroutine
end module

